### DISCUSSION TITLE
Dynamically add inputs to Action

### DISCUSSION THREAD

**Discussion by jswny** [community/community#25353] [Actions]
<p>Is there any way to dynamically add inputs to an Action? Or for the Action itself to access the environment variables defined in the workflow that it is being called from? I would like to use the same Action which requires different environment variables per app that I’m using them on, so I can’t define them all as inputs.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jswny">@jswny</a>,</p>
<p>You can try using the workflow command “<a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" rel="nofollow noopener"><strong>set-env</strong></a>” to create or update an environment variable for any steps running next in a job. All the subsequent steps in the same job have access to the new value.<br>
A simple demo:</p>
<pre><code class="lang-yaml">jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    env:
      MY_ENV: "env value 01"
    steps:
      - name: run action - 1st
        uses: actionOrg/actionRepo@master
        with:
          input: ${{ env.MY_ENV }}  # the input value is "env value 01"

      - name: update env
        run: echo "::set-env name=MY_ENV::env value 02"

      - name: run action - 2nd
        uses: actionOrg/actionRepo@master
        with:
          input: ${{ env.MY_ENV }}  # the input value is "env value 02"
</code></pre>
<p>when using the <strong>set-env</strong> command, you also can call the value of another variable to set the new value of the environment variable.<br>
For example:</p>
<pre><code class="lang-yaml">- name: update env
  run: |
    number=02
    echo "::set-env name=MY_ENV::env value $number"
   # the new value is "env value 02"
</code></pre>
